app/scripts/controllers/metametrics.js

```diff
@@ -581,6 +584,28 @@ export default class MetaMetricsController {
     }, {});
   }
 
+  /**
+   *
+   * @param {*} metamaskState
+   * @returns number of unique collectible addresses
+   */
+  _getNumberOfNFtCollection(metamaskState) {
```

Comment:
I wonder if this function is a good candidate for lodash's `memoize` method.

Reply:
You mean memoize the whole function or some parts of it?

Reply:
The whole function in which we'd pass only `allCollectibles` to the function. If the input is the same, it would return the same computed value and not need to recompute the function.

Reply:
On second thought, we could use `memoize` to compute `allAddresses`. 

What I'm thinking is that `allAddresses.length` will give us the `number_of_nfts` based on the description in the issue (please double-check me here):

> All collectibles has this shape:
allCollectibles: { [key: string (account)]: { [key: string (chainId)]: Collectible[] } };

> This is an object whose keys are accounts and the values are objects where the keys are chainIds. So, for example, in order to get an aggregate number of NFTs imported you'll have to go through each chainId in each account in this object and sum the length of the Collectibles array.

> In order to get the number of collections you'll need to do the same sort of thing but you'll want the number of unique 'address' properties on the entries. Each collectible has an address property that indicates which collection it is a part of.

Then maybe we can share `allAddresses` for both `number_of_nfts` and `number_of_nft_collections`. Let me know if this makes sense. Always up for hopping on a call as well.